src/devices/Ak8963/Ak8963.cs

```diff
@@ -0,0 +1,258 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers.Binary;
+using System.Device.I2c;
+using System.Numerics;
+using System.Threading;
+
+namespace Iot.Device.Ak8963
+{
+    public class Ak8963 : IDisposable
+    {
+        private I2cDevice _i2cDevice;
+        private bool _autoDispose;
+        private MeasurementMode _measurementMode;
+        private OutputBitMode _outputBitMode;
+        private bool _selfTest = false;
+        private Vector3 _magnetometerBias = new Vector3();
+        private Ak8963Interface _ak8963Interface;
+
+        public const byte DefaultI2cAddress = 0x0C;
+
+        public Ak8963(I2cDevice i2CDevice, bool autoDispose = true) : this(i2CDevice, new Ak8963I2c(), autoDispose)
+        { }
+
+        public Ak8963(I2cDevice i2CDevice, Ak8963Interface ak8963Interface, bool autoDispose = true)
+        {
+            _i2cDevice = i2CDevice;
+            _autoDispose = autoDispose;
+            _ak8963Interface = ak8963Interface;
+            // Initialize the default modes
+            _measurementMode = MeasurementMode.PowerDown;
+            _outputBitMode = OutputBitMode.Output14bit;
+            byte mode = (byte)((byte)_measurementMode | ((byte)_outputBitMode << 4));
+            WriteRegister(Register.CNTL, mode);
+        }
+
+        /// <summary>
+        /// Reset the device
+        /// </summary>
+        public void Reset()
+        {
+            WriteRegister(Register.RSV, 0x01);
+            Thread.Sleep(100);
```

Comment:
Why is sleep needed? is it in the spec?

Reply:
This register is not really documented. Documentation says the board is ready after 50 milliseconds. I can adjust to this value and leave a note

Reply:
please make it a const and add some comment that it's poorly documented and add a comment above the constant where 100/50 is from